pandas/core/indexes/base.py

```diff
@@ -1383,6 +1391,31 @@ def format(
 
         return self._format_with_header(header=header, na_rep=na_rep)
 
+    _default_na_rep = "NaN"
```

Comment:
Is this correct if the data is a masked array with NA?

Reply:
ATM Index.format and DTI.format have different defaults for na_rep.  Putting this here keeps the same behavior for _format_flat while avoiding the need for overrding the whole method in DTI.

Looks like na_rep is ignored inside format_with_header anyway

Reply:
> Looks like na_rep is ignored inside format_with_header anyway

That's probably not good, but I guess that can be tackled elsewhere